pkg/cmd/infra/router/template_test.go

```diff
@@ -0,0 +1,111 @@
+			expectError:        true,
+			expectErrorMessage: "invalid HTTP header input specification",
+		},
+		{
+			description: "should percent decode the multiple response header values",
+			inputValue:  "X-Frame-Options:DENY:Set,X-XSS-Protection:1%3Bmode%3Dblock:Set,x-forwarded-client-cert:%25%7B%2BQ%7D%5Bssl_c_der%2Cbase64%5D:Set",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "X-Frame-Options", Value: "'DENY'", Action: "Set"},
+				{Name: "X-XSS-Protection", Value: "'1;mode=block'", Action: "Set"},
+				{Name: "x-forwarded-client-cert", Value: "'%{+Q}[ssl_c_der,base64]'", Action: "Set"},
+			},
+			expectError: false,
+		},
+		{
+			description: "should percent decode the multiple request header values",
+			inputValue:  "Accept:text%2Fplain%2C+text%2Fhtml:Set,Accept-Encoding:Delete",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "Accept", Value: "'text/plain, text/html'", Action: "Set"},
+				{Name: "Accept-Encoding", Action: "Delete"},
+			},
+			expectError: false,
+		},
+		{
+			description:        "should not percent decode the multiple double encoded header values",
+			inputValue:         "X-Frame-Options%3ADENY%3ASet%2CX-Cache-Info%3Anot%2Bcacheable%253B%2Bmeta%2Bdata%2Btoo%2Blarge%3ASet%2CX-XSS-Protection%3ADelete%2CX-Source%3A%2525%255Bres.hdr%2528X-Value%2529%252Clower%255D%3ASet",
+			expectError:        true,
+			expectedValue:      nil,
+			expectErrorMessage: "invalid HTTP header input specification",
+		},
+		{
+			description: "should percent decode the multiple header values with simple non encoded strings",
+			inputValue:  "X-Frame-Options:DENY:Set",
+			expectedValue: []templateplugin.HTTPHeader{
+				{Name: "X-Frame-Options", Value: "'DENY'", Action: "Set"},
+			},
+			expectError: false,
+		},
+		{
+			description:        "When Action is `Set` it should not percent decode the incorrectly encoded value part Eg: percent is not followed by two hexadecimal digits",
+			inputValue:         "x-forwarded-client-cert:%25%7B%2BQ%7D%5Bssl_c_der%2Cbase64%5:Set",
+			expectError:        true,
+			expectedValue:      nil,
+			expectErrorMessage: "failed to decode percent encoding",
+		},
+		{
+			description:        "When Action is `Delete` it should not percent decode the incorrectly encoded value part Eg: percent is not followed by two hexadecimal digits",
+			inputValue:         "Accept-Encoding:Delete%2",
+			expectError:        true,
+			expectedValue:      nil,
+			expectErrorMessage: "failed to decode percent encoding: Delete%2",
+		},
+		{
+			description:        "should not allow specifying headers with no action or value",
+			inputValue:         "X-Frame-Options",
+			expectError:        true,
+			expectErrorMessage: "invalid HTTP header input specification: ",
+		},
+		{
+			description:        "should not allow blank values for headers",
+			inputValue:         "",
+			expectedValue:      nil,
+			expectError:        true,
+			expectErrorMessage: "encoded header string not present",
+		},
```

Comment:
Add a test for an Action that is not allowed - like something misspelled, or even `Append`.

Reply:
done